skimage/util/_backends.py

```diff
@@ -0,0 +1,127 @@
+import functools
+from importlib.metadata import entry_points
+from functools import lru_cache
+import os
+import warnings
+
+
+def _entry_points(group):
+    # Support Python versions before 3.10, which do not let you
+    # filter groups directly.
+    all_entry_points = entry_points()
+    if hasattr(all_entry_points, "select"):
+        selected_entry_points = all_entry_points.select(group=group)
+    else:
+        selected_entry_points = all_entry_points.get(group, ())
+    return selected_entry_points
+
+
+def disable_dispatching():
+    """Determine if dispatching has been disabled by the user"""
+    no_dispatching = os.environ.get("SKIMAGE_NO_DISPATCHING", False)
+    if no_dispatching == "1":
+        return True
+    else:
+        return False
+
+
+@lru_cache
```

Comment:
I don't think we should put a `@lru_cache` decorator here, because I think it checks if the input to the function are same and then returns the cached result. And this function doesn't take any inputs. So, if someone tries to run an algorithm without their backend installed the output of this function will get cached and then they will install their backend but that won't be reflected in the output of this function because it will use the cached result.

Reply:
`lru_cache` is only valid during runtime. So this would only be a problem if we expect the available backends to change during runtime. That's probably unlikely. 

However, from a glance, it seems that there's no real need to cache this function (yet)?

Reply:
The cache is indeed reset each time you start your program. Installing a backend after your program has started running seems like a very very niche use-case.

The reason I've added it is that it takes "quite a while" for the entrypoint mechanism to discover entrypoints. And `all_backends()` is called for every function that is decorated, so a lot of repeat calls.

An alternative to `lru_cache` would be a pattern like:
```python

def all_backends():
  # code here

_ALL_BACKENDS = all_backends()
```
And then using `_ALL_BACKENDS` in the decorator. For me that seems like a more "home brew"/DIY way of doing the same thing as `lru_cache`.

Maybe it is worth adding a comment about the use of `lru_cache` to the code to explain this.

Reply:
Could use `functools.cache` since the `lru_` part is meaningless and maybe confusing, but seems good to me (unless there is a cuter pattern suggested, I don't know it).  One can always micro-optimize later if it turns out a larger chunk if import is spend here.

This seems like one nice thing about using entry-points: we know all possible backends from the start (even if some of those backends may require additional steps to ever used).

Reply:
I find both of your arguments, @betatim, @seberg, convinving. :+1:

Reply:
TIL about `functools.cache`! Switched to it.

`cache` was added in Python 3.9, so I think we can use it/what is the oldest version of Python that scikit-image supports?

Reply:
As of 13 hours ago 3.10. See https://github.com/scikit-image/scikit-image/pull/7561. So, safe to use. :grin: